enhancements/single-node-developer-cluster-profile.md

```diff
@@ -0,0 +1,514 @@
+optimize the configuration of the individual components for single
+node developer clusters. In general, each component will be addressed
+independently and possibly multiple times, first to eliminate the
+post-install patching by SNC, and then to enable additional
+optimizations. During implementation of this phase, it is expected
+that patterns will emerge when common approaches apply across
+different components.
+
+### Risks and Mitigations
+
+One risk is that all parts of the implementation of the cluster
+profiles
+[enhancement](https://github.com/openshift/enhancements/blob/master/enhancements/update/cluster-profiles.md)
+do not get merged in a timely manner. Implementing the
+single-node-developer cluster profile requires the ability to specify
+a non-default cluster profile to the OpenShift installer, and for this
+to result in only the manifests annotated for the specified cluster
+profile to be processed by the resulting cluster. Developer tools team
+members have proposed CVO patches as part of that enhancement, and
+will continue to participate in the review process, but progress
+getting the implementation merged has been slow so far. Having this
+enhancement as an additional use case for the cluster profile
+mechanism hopefully will help clarify how that feature is being used
+and expedite its implementation.
+
+Another risk is that it will take too long to get the initial
+annotations merged to all the needed OpenShift component repositories
+during the first phase of development. Until all needed annotations
+are merged, the SNC tooling cannot be updated to make use of the
+single-node-developer cluster profile, as needed components would be
+excluded from the cluster. The review process for this enhancement
+should help raise awareness across the various teams that will need to
+review and merge these patches.
+
+A final risk is that new components will be added to OpenShift after
+the completion of phase 1 that should be included in single node
+developer clusters, but will not have the necessary annotation
+included in their manifests due to lack of awareness or concern for
+the CRC product. Given that single node developer clusters are
+expected to pass the majority of the OpenShift conformance tests,
+component developers should include this profile's annotation unless
+the new component is clearly not relevant to the development of
+applications on single node clusters. Developer tools team members
+will be available to collaborate with component developers regarding
+whether a component should be included in the single-node-developer
+cluster profile, and if so, how it might be configured to minimize
+resource requirements.
+
+## Design Details
+
+### Open Questions
+
+One currently open question is how to manage the set of operators
+visible within the operator hub on a single-node-developer
+cluster. Operators that cannot run without OpenShift components
```

Comment:
I think the ability to install and use operators from the operator hub is considered a key feature of CRC, so I doubt we can disable this entirely. I had a discussion with @ecordell last week to start exploring our options here. It sounds like OLM currently will prevent installing an operator that declares dependencies on APIs that aren't available, so that should help. Even if we don't come up with a comprehensive solution right away, this enhancement doesn't make things any worse than with the current CRC product.

Reply:
@rkukura i tend to agree with @soltysh .

having the catalog show function that is known to work by default (which would be empty, or a known subset that _CRC_ says is important to the desired developer story is preferred.

we can document or add some `crc` command line to show the full catalog if _CRC_ was being used as demo-ware to show ecoystem.

showing the whole catalog by default goes against the principle of "explicit inclusion".